src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py

```diff
@@ -296,6 +296,11 @@ def execute(self, emulator=None):
         sysTime = getrusage(RUSAGE_CHILDREN).ru_stime - sysTime0
         endTime = time.time()
         returnMessage = None
+        try:
+            # Update the job report with the relevant WMCMSSWSubprocess metrics
+            self.report.updateSubprocessInfo(sysTime, userTime, startTime, endTime)
+        except Exception as ex:
+            logging.error("Error updating job report with WMCMSSWSubprocess metrics: %s", str(ex))
```

Comment:
I am not 100% sure, but i fear that here we do not have access to our comp python environment, so we risk that self.report.updateSubprocessInfo is not defined. 

I would move these lines after: 

```python
    envOverride['PYTHONPATH'] = pythonPath
    os.environ.update(envOverride)
```

does it make sense to you? have you tested your changes running a job interactively and already can prove me wrong? :)

Reply:
successfull as you checked and the information is there or as "the jobs did not fail"? the try-except is there not to make anything crash.

in any case, since we are close to change python version and how libraries are loaded may change, i would be on the safe side and move these lines below, i do not want to have bad surprises a few weeks from now :)